File: 스택,큐/기능개발.py
import math
from collections import deque

def solution(progresses, speeds):
    answer = []
    days = [0] * len(progresses)
    for p in range(len(progresses)):
        days[p] = math.ceil((100 - progresses[p]) / speeds[p])
    print(days)
    # tc1: [7, 3, 9]
    # tc2: [5, 10, 1, 1, 20, 1]
    
    days_q = deque(days)
    print(days_q)
    cnt = 1
    while days_q:
        day = days_q.popleft()
        
        if days_q and day >= days_q[0]:
            cnt += 1
        else:
            answer.append(cnt)
            cnt = 1
    print(answer)
    return answer
solution([95, 90, 99, 99, 80, 99],[1, 1, 1, 1, 1, 1])

File: 스택,큐/기능개발.js
//https://programmers.co.kr/learn/courses/30/lessons/42586?language=javascript
function solution(progresses, speeds) {
    var answer = [];
    let length = progresses.length;
    let s = -1
    let cur = 0
    while (cur !=length) {
        s = parseInt(( 100 - progresses[cur] ) / speeds[cur]) < ( 100 -  progresses[cur] ) / speeds[cur] 
        ? parseInt(( 100 -  progresses[cur] ) / speeds[cur])+1 : parseInt(( 100 -  progresses[cur] ) / speeds[cur]) 
        let i = cur;
        for(;i<length;i++){
            progresses[i] += (speeds[i] * s)
        }
        let cnt = 0 
        let tmp = progresses.slice(cur)

        for(let i =0; i <tmp.length;i++){
            if(tmp[i]<100) break 
            cnt += 1
        }
        cur += cnt
        answer.push(cnt)
    }
    return answer;
}
console.log(solution([93,30,55],[1,30,5]));